src/controllers/addressBook/addressBook.test.ts

```diff
@@ -18,37 +23,55 @@ const MOCK_ACCOUNTS: Account[] = [
     addr: '0x598cD170E9b90e9c7E57e18B47D589ceC119744c',
     associatedKeys: [],
     initialPrivileges: [],
-    creation: null
+    creation: null,
+    preferences: {
+      label: DEFAULT_ACCOUNT_LABEL,
+      pfp: '0x598cD170E9b90e9c7E57e18B47D589ceC119744c'
+    }
   },
   {
     addr: '0x66fE93c51726e6FD51668B0B0434ffcedD604d08',
     associatedKeys: [],
     initialPrivileges: [],
-    creation: null
+    creation: null,
+    preferences: {
+      label: 'Account 1',
+      pfp: '0x66fE93c51726e6FD51668B0B0434ffcedD604d08'
+    }
   },
   {
     addr: '0x31800a810A2d9C3315dc714e1Eb988bd6A641eF0',
     associatedKeys: [],
     initialPrivileges: [],
-    creation: null
+    creation: null,
+    preferences: {
+      label: 'Account 2',
+      pfp: '0x31800a810A2d9C3315dc714e1Eb988bd6A641eF0'
+    }
   }
 ]
 
-const MOCK_ACCOUNT_PREFERENCES: AccountPreferences = {
-  '0x66fE93c51726e6FD51668B0B0434ffcedD604d08': {
-    label: 'Account 1',
-    pfp: '0x66fE93c51726e6FD51668B0B0434ffcedD604d08'
-  },
-  '0x31800a810A2d9C3315dc714e1Eb988bd6A641eF0': {
-    label: 'Account 2',
-    pfp: '0x31800a810A2d9C3315dc714e1Eb988bd6A641eF0'
-  }
-}
+storage.set('accounts', MOCK_ACCOUNTS)
+
+const providers = Object.fromEntries(
+  networks.map((network) => [network.id, getRpcProvider(network.rpcUrls, network.chainId)])
+)
 
 describe('AddressBookController', () => {
-  const addressBookController = new AddressBookController(storage, MOCK_ACCOUNTS, {
-    accountPreferences: MOCK_ACCOUNT_PREFERENCES
-  } as SettingsController)
+  let providersCtrl: ProvidersController
+  const networksCtrl = new NetworksController(
+    storage,
+    (net) => {
```

Comment:
Note: if https://github.com/AmbireTech/ambire-common/pull/815 gets merged first, need to pass the `fetch` as param after the `storage` here.